app/assets/javascripts/discourse/app/form-kit/components/fk/object.gjs

```diff
@@ -9,65 +9,55 @@ export default class FKObject extends Component {
   }
 
   get name() {
-    return this.args.parentName
-      ? `${this.args.parentName}.${this.args.name}`
-      : this.args.name;
-  }
-
-  get keys() {
-    return Object.keys(this.objectData).map((name) => {
-      return { name };
-    });
-  }
-
-  entryData(name) {
-    return this.objectData[name];
+    return this.args.name
+      ? `${this.args.parentName ? this.args.parentName + "." : ""}${
+          this.args.name
+        }`
+      : this.args.parentName;
   }
 
   <template>
     <div class="form-kit__object">
-      {{#each this.keys key="name" as |key|}}
```

Comment:
This was trying to be too smart IMO. We were iterating over keys for the dev, but I think it's better as responsibility of the developer to decide which keys to use here.

Reply:
@jjaffeux - This change is causing children fields to be registered twice on updating a field. I suspect it's because we were yielding different instances of the components.

Reply:
Indeed this is why I had added key="name" before